### PR TITLE
initial integration with Google Drive and Sheets API

### PR DESCRIPTION
**Pull request by binwei**
What is included:
-- implementation for photo upload and spreadsheet update
-- gapi_cred.json is NOT part of the files committed

The resources used by this implementation include
-- Google Drive folder for storing the uploaded images
-- Google Spreadsheet for storing the registration fields

These resource IDs are hardcoded for now but need to be passed through env.

### CODE REVIEW

**Comment by binwei** [comment]
I refactored resource IDs and gapi credentials into a dot env file, such that these resource IDs are passed through process.env.

File: server.js [hunk 1]
```diff
@@ -6,6 +6,34 @@ const app = express();
 // for photo upload from frontend
 const upload = multer();
 
+let google = require('googleapis');
```
Comment (tonyhualiu) [inline server.js]: use `const` for immutable variable.

File: server.js [hunk 2]
```diff
@@ -6,6 +6,34 @@ const app = express();
...
+
+// configure a JWT auth client
+var private_key_value = process.env.gapi_private_key.replace(/\\n/g, '\n');
```
Comment (tonyhualiu) [inline server.js]: use `const`

Comment (tonyhualiu) [reply server.js]: same as the `jwtClient`, `drive`, `sheets`

File: server.js [hunk 3]
```diff
@@ -25,6 +53,67 @@ app.post("/api/form", (req, res) => {
...
+    // How the input data should be interpreted.
+    valueInputOption: 'USER_ENTERED',
+    // How the input data should be inserted.
+    insertDataOption: 'INSERT_ROWS',
+    auth: jwtClient,
+    resource: resourceValues
+  }, function(err, response) {
+    if (err) {
+      console.error(err);
+      return;
```
Comment (tonyhualiu) [inline server.js]: please add TODO to return server error to frontend

File: server.js [hunk 4]
```diff
@@ -25,6 +53,67 @@ app.post("/api/form", (req, res) => {
...
+    }
+
+    console.log(JSON.stringify(response, null, 2));
+  });
+
+  // TODO upon successful submission, rename the file and move the image to private drive folder
+  // TODO how do we propagate file id from the previous post
```
Comment (tonyhualiu) [inline server.js:93]: NO_ACTION but we definitely should talk about this

File: server.js [hunk 5]
```diff
@@ -39,7 +128,32 @@ app.post("/api/photo", upload.single("avatar"), (req, res) => {
...
   // uploaded image.
-  res.json({url: 'http://www.sandiegovips.com/wp-content/uploads/2014/05/test-image.jpeg'});
+
+  const fileMetadata = {
+    name: req.file.originalname,
+    parents: [process.env.gapi_public_folder_id]
+  };
+
+  const media = {
+    mimeType: 'image/jpeg',
```
Comment (tonyhualiu) [inline server.js]: can you add TODO here to handle other formats of images?

File: server.js [hunk 6]
```diff
@@ -39,7 +128,32 @@ app.post("/api/photo", upload.single("avatar"), (req, res) => {
...
+
+  drive.files.create({
+    auth: jwtClient,
+    resource: fileMetadata,
+    media,
+    fields: 'id'
+  }, (err, file) => {
+    if (err) {
+      console.log(err);
+      return;
```
Comment (tonyhualiu) [inline server.js]: TODO: return an error to the frontend.

**Comment by tonyhualiu** [comment]
Can you please address the comments? Other than those, it looks good.

**Comment by binwei** [comment]
Please review again. I added more TODO's.

**Pull request merged**